File: solutions/java/isbn-verifier/1/src/main/java/IsbnVerifier.java
class IsbnVerifier {

    boolean isValid(String stringToVerify) {
        int currentMultiplier = 10;
        int sum = 0;
        int digitCount = 0;
        for (char ch: stringToVerify.toCharArray()) {
            if (ch == '-') continue;
            if (currentMultiplier == 0) return false;

            digitCount += 1;

            if (digitCount < 10 && ch == 'X') return false;
            
            sum += this.getDigit(ch) * currentMultiplier;
            currentMultiplier -= 1;            
        }

        return digitCount == 10 && sum % 11 == 0;
    }

    private int getDigit(char ch) {
        if (ch == 'X') return 10;
        return ch - '0';
    }

}

File: solutions/cpp/isbn-verifier/1/isbn_verifier.cpp
#include "isbn_verifier.h"

namespace isbn_verifier {
    bool is_valid(const string& candidate) {
        int i{};
        int sum{};
        for (auto it = candidate.rbegin(); it != candidate.rend(); ++it) {
            char ch = *it;
            if (ch == '-') {
                continue;
            }

            int digit = ch - '0';
            if (ch == 'X' && i == 0) {
                digit = 10;
            }
            sum += digit * (i + 1);
            ++i;
        }
        return i == 10 && sum % 11 == 0;
    }
}  // namespace isbn_verifier